Author: "Ferenc Wagner" <wferi@niif.hu>
Description: Fix for clearing DOT_FONTNAME.

--- a/docs/man.dox.in
+++ b/docs/man.dox.in
@@ -135,7 +135,7 @@
 MSCGEN_PATH            =
 HIDE_UNDOC_RELATIONS   = YES
 HAVE_DOT               = NO
-DOT_FONTNAME           = FreeSans
+DOT_FONTNAME           =
 DOT_FONTSIZE           = 10
 DOT_FONTPATH           =
 CLASS_GRAPH            = NO
--- a/docs/html.dox.in
+++ b/docs/html.dox.in
@@ -185,7 +185,7 @@
 MSCGEN_PATH            =
 HIDE_UNDOC_RELATIONS   = YES
 HAVE_DOT               = YES
-DOT_FONTNAME           = FreeSans
+DOT_FONTNAME           =
 DOT_FONTSIZE           = 10
 DOT_FONTPATH           =
 CLASS_GRAPH            = YES
