Author: "Ferenc Wagner" <wferi@niif.hu>
Description: Fix for comment typo, incomming.

--- a/include/qb/qbipcs.h
+++ b/include/qb/qbipcs.h
@@ -237,7 +237,7 @@
 void qb_ipcs_destroy(qb_ipcs_service_t* s);
 
 /**
- * Limit the incomming request rate.
+ * Limit the incoming request rate.
  * @param s service instance
  * @param rl the new rate
  */
@@ -245,7 +245,7 @@
 			       	enum qb_ipcs_rate_limit rl);
 
 /**
- * Send a response to a incomming request.
+ * Send a response to a incoming request.
  *
  * @param c connection instance
  * @param data the message to send
@@ -260,7 +260,7 @@
 			      size_t size);
 
 /**
- * Send a response to a incomming request.
+ * Send a response to a incoming request.
  *
  * @param c connection instance
  * @param iov the iovec struct that points to the message to send
