Author: "Ferenc Wagner" <wferi@niif.hu>
Description: Fix for qbutil descriptioin.

--- a/include/qb/qbutil.h
+++ b/include/qb/qbutil.h
@@ -36,10 +36,10 @@
 
 /**
  * @file qbutil.h
- * @author Angus Salkeld <asalkeld@redhat.com>
- *
  * These are some convience functions used throughout libqb.
  *
+ * @author Angus Salkeld <asalkeld@redhat.com>
+ *
  * @par Locking
  * - qb_thread_lock_create()
  * - qb_thread_lock()
